### PR TITLE
CI Improvements

### PR DESCRIPTION
**Pull request by Drakojin**
- Optimized Gradle scripts
- Implemented IntelliJ Code Inspection in CI
- Stages for CI 

refer #101

### CODE REVIEW

File: .idea/scopes/livingdoc2.xml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<component name="DependencyValidationManager">
+  <scope name="livingdoc2" pattern="(test:*..*||src:*..*)&amp;&amp;!src[group:livingdoc-documentation]:*..*" />
```
Comment (pkleimann) [inline .idea/scopes/livingdoc2.xml]: Better is to not use '2' in the name. Only name="livingdoc".

File: .ideainspect [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+# This is _the prefered way_ to limit your inspection run to a part of your project files
+# as it takes effect within IDEA itself.
+# See: https://www.jetbrains.com/help/idea/2016.2/specify-inspection-scope-dialog.html
+#
+# HOWTO:
+# 1) Create a new scope excluding undesired folders/files (node_modules, doc, ...)
+# 2) Share the .idea/scopes/scopename.xml with the project
+# 3) Use the _name_ of the scope (not the file).
+# 4) Stick to a single word for best compability
+scope: livingdoc2
```
Comment (pkleimann) [inline .ideainspect]: scope: livingdoc

File: .travis.yml [hunk 3]
```diff
@@ -1,10 +1,35 @@
...
 
-script:
- - ./gradlew build
+before_cache:
+  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+
+jobs:
+ include:
+    - stage: testandvalidate
```
Comment (pkleimann) [inline .travis.yml]: here stage 'test' which is default

File: .travis.yml [hunk 4]
```diff
@@ -1,10 +1,35 @@
...
+      jdk: openjdk8
+      script: ./gradlew check
+    - jdk: oraclejdk8
+      script: ./gradlew check
+    - jdk: oraclejdk8
```
Comment (pkleimann) [inline .travis.yml]: here stage 'validate' or 'inspect'

File: .travis.yml [hunk 5]
```diff
@@ -1,10 +1,35 @@
...
+      env: validate
```
Comment (pkleimann) [inline .travis.yml]: if stage is named 'validate' do we need an env: validate anymore?

File: .travis.yml (hunk 3 above)
Comment (Drakojin) [reply .travis.yml]: test and validate would better describe it right now
the gradle check tasks (as mentioned below) just run junit5 tests
the third one (for speed reasons) just runs the static code analysis

File: .travis.yml (hunk 5 above)
Comment (Drakojin) [reply .travis.yml]: just a marker, as it is the only one of the three parallel tasks that runs code analysis
can be removed if not needed

**Pull request merged**